### PR TITLE
Add documentation about the identity federation concept in SCS

### PR DESCRIPTION
**Pull request by reqa**
Addressing https://github.com/SovereignCloudStack/issues/issues/397

### CODE REVIEW

File: operations/iam/openstack-federation-via-oidc.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+## 1. Keystone
+
+[Keytone supports federated identities](https://docs.openstack.org/keystone/latest/admin/federation/federated_identity.html).
+To allow SCS to consume identities managed in external IAM solutions,
+federation protocols like OpenID Connect or SAML can be used.
+Keystone currently makes use of third party apache modules like
+[mod\_auth\_openidc](https://github.com/OpenIDC/mod_auth_openidc),
+[mod\_oauth2](https://github.com/OpenIDC/mod_oauth2) and
+[mod\_auth\_mellon](https://github.com/UNINETT/mod_auth_mellon) to delegate
+authentication to an SSO IdP (i.e. SAML IdP or OpenID Connect provider).
```
Comment (JuanPTM) [inline operations/iam/openstack-federation-via-oidc.md:26]:
```suggestion
authentication to a SSO IdP (i.e. SAML IdP or OpenID Connect provider).
```

File: operations/iam/openstack-federation-via-oidc.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+### 1.2 Keystone mapping of token claims
+
+Upon login of a user Keystone evaluates the credentials obtained from the ID token that the IdP issued.
+These include group memberships and roles, which can be used to assign the user to a certain project.
+Keystone maps these external identities to internal (shadow) users.
+It can either attempt to map the obtained information to a `local` type user, which needs to be
+provisioned before authentication by external tooling, or it can be instructed via the mapping to
+generate an `ephemeral` type user. `ephemeral` users are cleaned up automatically after some time
+of inactivity and with that, they lose access to projects, iff the access is granted indirectly
```
Comment (JuanPTM) [inline operations/iam/openstack-federation-via-oidc.md:62]:
```suggestion
of inactivity and with that, they lose access to projects, if the access is granted indirectly
```

File: operations/iam/openstack-federation-via-oidc.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+which is the currency that is understood by other OpenStack services and can be used to access them.
+
+In SCS we want to represent each customer by a sepatate dedicated OpenStack domain to host
+their projects and (shadow) user accounts.
+
+The processing of information from the OIDC tokens is configured by two parts. The first part is the
+`rules.json` mapping which is described in the
+[OpenStack federation mapping combinations](https://docs.openstack.org/keystone/latest/admin/federation/mapping_combinations.html)
+document. This file is used to configure Keystones internal mapping engine
+and it needs to be attached to some OpenStack domain, which is named `keycloak` be default in SCS.
```
Comment (JuanPTM) [inline operations/iam/openstack-federation-via-oidc.md:78]:
```suggestion
and it needs to be attached to some OpenStack domain, which is named `keycloak` by default in SCS.
```

File: operations/iam/openstack-federation-via-oidc.md (hunk 2 above)
Comment (reqa) [reply operations/iam/openstack-federation-via-oidc.md:62]: No, I wanted `iff`, in the mathematical sense: If and only if.

**Review by garloff — APPROVED** [review]
This looks good to me.
I wonder whether this should become a set of ADRs in the standards repo though.

**Comment by fkr** [comment]
> This looks good to me. I wonder whether this should become a set of ADRs in the standards repo though.

yes, I did actually think the same when reading it.

**Comment by garloff** [comment]
Let's merge it for now -- tracking a possible move in #74.

**Pull request merged**